library/aur.py

```diff
@@ -145,39 +159,61 @@ def check_packages(module, packages):
     module.exit_json(changed=status, msg=message, diff=diff)
 
 
-def build_command_prefix(use, extra_args, skip_pgp_check=False, ignore_arch=False, aur_only=False):
+def build_command_prefix(use, extra_args, skip_pgp_check=False, ignore_arch=False, aur_only=False, local_pkgbuild=None):
     """
     Create the prefix of a command that can be used by the install and upgrade functions.
     """
-    command = def_lang + use_cmd[use]
+    if local_pkgbuild:
+        command = def_lang + use_cmd_local_pkgbuild[use]
+    else:
+        command = def_lang + use_cmd[use]
     if skip_pgp_check:
         command.append('--skippgpcheck')
     if ignore_arch:
         command.append('--ignorearch')
     if aur_only and use in has_aur_option:
         command.append('--aur')
+    if local_pkgbuild and use != 'makepkg':
+        command.append(local_pkgbuild)
     if extra_args:
         command += shlex.split(extra_args)
     return command
 
 
-def install_with_makepkg(module, package, extra_args, skip_pgp_check, ignore_arch):
+def install_with_makepkg(module, package, extra_args, skip_pgp_check, ignore_arch, local_pkgbuild=None):
+    """
+    Install the specified package or a local PKGBUILD with makepkg
+    """
+    if not local_pkgbuild:
+        module.get_bin_path('fakeroot', required=True)
+        f = open_url('https://aur.archlinux.org/rpc/?v=5&type=info&arg={}'.format(urllib.parse.quote(package)))
+        result = json.loads(f.read().decode('utf8'))
+        if result['resultcount'] != 1:
+            return (1, '', 'package {} not found'.format(package))
+        result = result['results'][0]
+        f = open_url('https://aur.archlinux.org/{}'.format(result['URLPath']))
+    with tempfile.TemporaryDirectory() as tmpdir:
+        if local_pkgbuild:
+            copy_tree(local_pkgbuild, tmpdir)
```

Comment:
why not the regular `copytree()` ?

Reply:
I get ```FileExistsError: [Errno 17] File exists: '/tmp/tmp_bkpq7iv'``` with ```copytree```. There are no issues with ```copy_tree```..

Reply:
The destination directory must not already exist. it behaves differently.

Reply:
ok. this works.